### PR TITLE
Cpo 1914 arizona taxes

### PR DESCRIPTION
**Pull request by cstait**
Was unsure how to handle this one in a similar vein to others, as I thought the best way to handle Arizona's percentages would be as an enum, but I was unsure how to abstract enums so I just stuck with one class.

### CODE REVIEW

**Comment by cstait** [comment]
ah I see your comment now, will do.

File: CertiPay.Taxes.State.Tests/AZ/TaxTable2017Tests.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    [TestFixture]
+    public class TaxTable2017Tests
+    {       
+         
+        [Test]        
+        [TestCase(1500, Arizona.TaxRate.fiveone, 76.5)]      
+        public void AZ_2017_Checks_And_Balances(decimal grossWages, Arizona.TaxRate rate, decimal expected)
+        {
+            var taxTable = TaxTables.GetForState(StateOrProvince.AZ, 2017) as Arizona.TaxTable2017;
```
Comment (mattgwagner) [inline CertiPay.Taxes.State.Tests/AZ/TaxTable2017Tests.cs]: Better practice here would be to only "expect" it to return some specific implementation `Arizona.TaxTable`, but same expectations.

File: CertiPay.Taxes.State/Arizona/TaxTable2017.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+namespace CertiPay.Taxes.State.Arizona
+{
+    public  class TaxTable2017 : TaxTableHeader
+    {
+        public override StateOrProvince State { get; internal set; } = StateOrProvince.AZ;
+        public override int Year { get { return 2017; } }
+        public override Decimal SUI_Wage_Base { get { return 7000; } }        
+        public virtual Decimal Calculate(Decimal grossWages, TaxRate taxRate)
+        {
+            return grossWages * GetTaxRate(taxRate);
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/Arizona/TaxTable2017.cs]: Where does this tax rate selection come from? What does it actually represent?

File: CertiPay.Taxes.State/Arizona/TaxTable2017.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        public decimal GetTaxRate(TaxRate taxRate)
+        {
+            if (taxRate == TaxRate.zeroeight)
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/Arizona/TaxTable2017.cs]: For enums, we have this amazing technique called a `switch` statement in C# 😉

File: CertiPay.Taxes.State/Arizona/TaxTable2017.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                return 0.051m;
+            else
+                return 0.00m;                 
+            }
+        }
+
+
+        public enum TaxRate : Byte
+        {
+            [Display(Name = "0.080%")]
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/Arizona/TaxTable2017.cs]: Is this actually 0.080% or actually 0.8%? Same below.

File: CertiPay.Taxes.State/Arizona/TaxTable2017.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace CertiPay.Taxes.State.Arizona
+{
+    public  class TaxTable2017 : TaxTableHeader
+    {
+        public override StateOrProvince State { get; internal set; } = StateOrProvince.AZ;
```
Comment (mattgwagner) [inline CertiPay.Taxes.State/Arizona/TaxTable2017.cs]: This calculation seems generally unchanged for a few years now -- look at some of the other states and see if we can make this more general and not tied only to 2017 (so we have less updates to do in 2018).

File: CertiPay.Taxes.State/Arizona/TaxTable2017.cs (hunk 2 above)
Comment (cstait) [inline CertiPay.Taxes.State/Arizona/TaxTable2017.cs]: as far as I can tell from the documentation, Employees choose their percentage rate? Really only gives guidelines for what qualifies as exempt:


EE Instructions:
https://www.azdor.gov/Portals/0/ADOR-forms/TY2017/10100/10121_f.pdf

Employer Instructions:
https://www.azdor.gov/Portals/0/ADOR-forms/TY2017/10100/10121_inst.pdf

**Comment by mattgwagner** [comment]
Changes look good -- I did a bit of background checking, and employees pick the rate that corresponds to the level of taxes they expect to pay based on a table. The amount of taxes they pay is a percentage of the FWT they pay for that year.

**Pull request merged**